de.metas.contracts/src/main/java/de/metas/contracts/order/ContractOrderRepository.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * 
+ */
+package de.metas.contracts.order;
+
+import java.util.List;
+
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.adempiere.util.proxy.Cached;
+import org.springframework.stereotype.Repository;
+
+import de.metas.contracts.model.I_C_Flatrate_Term;
+import de.metas.contracts.subscription.model.I_C_Order;
+import de.metas.contracts.subscription.model.I_C_OrderLine;
+import de.metas.order.OrderId;
+import de.metas.util.Services;
+import lombok.NonNull;
+
+/*
+ * #%L
+ * de.metas.contracts
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+@Repository
+public class ContractOrderRepository
+{
+	@Cached(cacheName = I_C_Flatrate_Term.Table_Name + "#by#OrderId")
+	public List<I_C_Flatrate_Term> retrieveFlatrateTerms(@NonNull final OrderId orderId)
```

Comment:
* i suggest the following convention: if it records "records" as is the case here, then call it "..DAO".
* in this particular case, we have already `IContractsDAO` *and* `IFlatrateDAO` that return terms. adding a third class doing the same thing..i think that's just making the code more complicated
* in this particular case i see two options:
* don't return a I_C_Flatrate_Term, but an "entity" pojo with the properties you actually need (this is the preferred one)
* add the method to IContractsDAO